include/EVT/io/platform/f3xx/f302x8/ADCf302x8.hpp

```diff
@@ -0,0 +1,84 @@
+#ifndef _EVT_ADCf302x8_
+#define _EVT_ADCf302x8_
+
+#include <HALf3/stm32f3xx.h>
+#include <HALf3/stm32f3xx_hal.h>
+
+#include <EVT/io/ADC.hpp>
+
+namespace EVT::core::IO {
+
+class ADCf302x8 : public ADC {
+public:
+
+    /**
+     * Setup the given pin for ADC usage
+     *
+     * @param pin The pin to setup for ADC
+     */
+    ADCf302x8(Pin pin);
+
+    /**
+     * Read the value on the ADC in volts.
+     *
+     * @return The voltage in volts
+     */
+    float read();
+
+    /**
+     * Read the raw value from the ADC
+     *
+     * @return The raw value from the ADC
+     */
+    uint32_t readRaw();
+
+    /**
+     * Read the value from the ADC as a percentage of the possible values
+     * from 0 to 1
+     *
+     * @return The ADC value as a percentage
+     */
+    float readPercentage();
+
+
+private:
+    // Max number of channels supported by the ADC
+    static constexpr uint8_t MAX_CHANNELS = 15;
+    // Max voltage of the ADC
+    static constexpr float MAX_VOLTAGE = 3.6;
```

Comment:
Curious this is 3.6? I would think 3.3, but if the datasheet says otherwise, obviously keep it as is.

Reply:
Yes I found this somewhere as the max, i can find the source if you would like. I think its based on the max logic high level the system has in spec

Reply:
Page 293 of the reference manual seems to have it.